packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/DynamicFromMap.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.bridge
+
+import androidx.core.util.Pools.SimplePool
+
+/** Implementation of Dynamic wrapping a ReadableMap. */
+internal class DynamicFromMap
+// This is a pools object. Hide the constructor.
+private constructor() : Dynamic {
+    private var mMap: ReadableMap? = null
+    private var mName: String? = null
+
+    override fun recycle() {
+        mMap = null
+        mName = null
+        sPool.get()?.release(this)
+    }
+
+    override val isNull: Boolean
+        get() {
+            check(!(mMap == null || mName == null)) { DYNAMIC_VALUE_RECYCLED_FAILURE_MESSAGE }
+            return mMap!!.isNull(mName!!)
+        }
+
+    override fun asBoolean(): Boolean {
+        check(!(mMap == null || mName == null)) { DYNAMIC_VALUE_RECYCLED_FAILURE_MESSAGE }
+        return mMap!!.getBoolean(mName!!)
+    }
+
+    override fun asDouble(): Double {
+        check(!(mMap == null || mName == null)) { DYNAMIC_VALUE_RECYCLED_FAILURE_MESSAGE }
+        return mMap!!.getDouble(mName!!)
+    }
```

Comment:
Using `!!` assumes the value is not null; is it guaranteed here? If possible, could you try an alternative to avoid using it?

Same for the rest.

Reply:
Right, theoretically these could be modified between the check and access. I will refactor this to use `checkNotNull`.